sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -210,6 +212,19 @@ object DataSourceV2Strategy extends Strategy with PredicateHelper {
     case AlterNamespaceSetProperties(catalog, namespace, properties) =>
       AlterNamespaceSetPropertiesExec(catalog, namespace, properties) :: Nil
 
+    case CommentOnNamespace(ResolvedNamespace(catalog, namespace), comment) =>
+      AlterNamespaceSetPropertiesExec(
+        catalog,
+        namespace,
+        Map(SupportsNamespaces.PROP_COMMENT -> comment)) :: Nil
+
+    case CommentOnTable(plan, comment) => plan match {
+      case ResolvedTable(catalog, identifier, _) =>
+        val changes = TableChange.setProperty(TableCatalog.PROP_COMMENT, comment)
+        AlterTableExec(catalog, identifier, Seq(changes)) :: Nil
+      case _ => throw new AnalysisException("COMMENT ON TABLE does not support views.")
```

Comment:
We shouldn't let planner to do this check for each command, how about we do the check as https://github.com/apache/spark/pull/26847/files#r361632741 ?

Reply:
ah. i missed that comment